apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/IgnoredTracerContext.java

```diff
@@ -34,14 +34,21 @@ public class IgnoredTracerContext implements AbstractTracerContext {
     private static final NoopSpan NOOP_SPAN = new NoopSpan();
     private static final String IGNORE_TRACE = "Ignored_Trace";
 
+    private LinkedList<AbstractSpan> activeSpanStack;
+
     private final CorrelationContext correlationContext;
     private final ExtensionContext extensionContext;
     private final ProfileStatusContext profileStatusContext;
 
-    private int stackDepth;
-
     public IgnoredTracerContext() {
-        this.stackDepth = 0;
+        this.activeSpanStack = new LinkedList<>();
+        this.correlationContext = new CorrelationContext();
+        this.extensionContext = new ExtensionContext();
+        this.profileStatusContext = ProfileStatusContext.createWithNone();
+    }
+
+    public IgnoredTracerContext(LinkedList<AbstractSpan> activeSpanStack) {
+        this.activeSpanStack = activeSpanStack;
```

Comment:
We only need the depth to be initialized based on stack depth, rather than holding all